feature/platform/tests/telemetry_inventory_test/telemetry_inventory_test.go

```diff
@@ -43,18 +43,49 @@ var componentType = map[string]oc.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT{
 	"Storage":     oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_STORAGE,
 }
 
+// validInstallComponentTypes indicates for each component type, which types of
+// install-component it can have (i.e., what types of components can it be installed into).
+var validInstallComponentTypes = map[oc.Component_Type_Union]map[oc.Component_Type_Union]bool{
+	oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_CONTROLLER_CARD: {
+		oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_CHASSIS: true,
+	},
+	oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_FABRIC: {
+		oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_CHASSIS: true,
+	},
+	oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_FAN: {
+		oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_CHASSIS:  true,
+		oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_FAN_TRAY: true,
```

Comment:
Note: we may discover we need to loosen this criteria.  I think it's ok to have it here for now and see what we discover.  (I am thinking about scenarios where a fan might be installed on a CPU, on a linecard, fabric card, etc.  Fans could be anywhere in a device.

Reply:
Sounds good!